dump/plugins/port.py

```diff
@@ -0,0 +1,112 @@
+import os, sys
+from .executor import Executor
+from dump.match_infra import MatchEngine, MatchRequest
+from dump.helper import display_template
+
+class Port(Executor):
+    
+    ARG_NAME = "port_name"
+    
+    def __init__(self):
+        self.match_engine = MatchEngine()
+          
+    def get_all_args(self, ns):
+        req = MatchRequest()
+        req.db = "CONFIG_DB"
+        req.table = "PORT"
+        req.key_pattern = "*"
+        req.ns = ns
+        ret = self.match_engine.fetch(req)
+        all_ports = []
+        for key in ret["keys"]:
+            all_ports.append(key.split("|")[-1])
+        return all_ports
+            
+    def execute(self, params_dict):
+        self.ret_temp = display_template(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        port_name = params_dict[Port.ARG_NAME]
+        self.ns = params_dict["namespace"]
+        self.get_config_info(port_name)
+        self.get_appl_info(port_name)
+        port_asic_obj = self.get_asic_info_hostif(port_name)
+        self.get_asic_info_port(port_asic_obj)
+        self.get_state_info(port_name)
+        return self.ret_temp
+    
+    def get_config_info(self, port_name):
```

Comment:
Changed the names, i would prefer modifying state instances instead of the pass by reference style you were asking about since, that would make the code cluttered. This'll be more evident when there are multiple levels of functions (Here it is just one), In the case of COPP, it is a mess passing the same object around. 

To make this a standard approach, i've decided to use state instance instead of pass by reference semantics